Library/ViewModels/CommentsViewModelTests.swift

```diff
@@ -7,9 +7,97 @@ import ReactiveSwift
+      .observe(self.configureCommentComposerViewIsBacking.observer)
+    self.vm.outputs.isCommentComposerHidden.observe(self.isCommentComposerHidden.observer)
+  }
+
+  func testOutput_ConfigureCommentComposerViewWithData_IsLoggedOut() {
+    self.configureCommentComposerViewURL.assertDidNotEmitValue()
+    self.configureCommentComposerViewIsBacking.assertDidNotEmitValue()
+
+    withEnvironment(currentUser: nil) {
+      self.vm.inputs.configureWith(project: .template)
+      self.vm.inputs.viewDidLoad()
+
+      self.configureCommentComposerViewURL
+        .assertValues([nil], "nil is emitted because the user is not logged in.")
+      self.configureCommentComposerViewIsBacking
+        .assertValues([false], "false is emitted because the project is not backed.")
+    }
+  }
+
+  func testOutput_ConfigureCommentComposerViewWithData_IsLoggedIn_IsBacking_False() {
+    let user = User.template |> \.id .~ 12_345
+
+    self.configureCommentComposerViewURL.assertDidNotEmitValue()
+    self.configureCommentComposerViewIsBacking.assertDidNotEmitValue()
+
+    withEnvironment(currentUser: user) {
+      self.vm.inputs.configureWith(project: .template)
+      self.vm.inputs.viewDidLoad()
+
+      self.configureCommentComposerViewURL
+        .assertValues(
+          [URL(string: "http://www.kickstarter.com/medium.jpg")],
+          "An URL is emitted because the user is logged in."
+        )
+      self.configureCommentComposerViewIsBacking
+        .assertValues([false], "false is emitted because the project is not backed.")
+    }
+  }
+
+  func testOutput_ConfigureCommentComposerViewWithData_IsLoggedIn_IsBacking_True() {
+    let project = Project.template
+      |> \.personalization.isBacking .~ true
+
+    let user = User.template |> \.id .~ 12_345
+
+    self.configureCommentComposerViewURL.assertDidNotEmitValue()
+    self.configureCommentComposerViewIsBacking.assertDidNotEmitValue()
+
+    withEnvironment(currentUser: user) {
+      self.vm.inputs.configureWith(project: project)
+      self.vm.inputs.viewDidLoad()
+
+      self.configureCommentComposerViewURL
+        .assertValues(
+          [URL(string: "http://www.kickstarter.com/medium.jpg")],
+          "An URL is emitted because the user is logged in."
+        )
+      self.configureCommentComposerViewIsBacking
+        .assertValues([true], "true is emitted because the project is backed.")
+    }
+  }
+
+  func testOutput_IsCommentComposerHidden_False() {
+    let user = User.template |> \.id .~ 12_345
+
+    withEnvironment(currentUser: user) {
+      self.vm.inputs.configureWith(project: .template)
+      self.vm.inputs.viewDidLoad()
+
+      self.isCommentComposerHidden.assertValue(false)
+    }
+  }
+
+  func testOutput_IsCommentComposerHidden_True() {
+    withEnvironment(currentUser: nil) {
+      self.vm.inputs.configureWith(project: .template)
+      self.vm.inputs.viewDidLoad()
+
+      self.isCommentComposerHidden.assertValue(true)
+    }
   }
```

Comment:
Can remove these because the conditions above cover the three scenarios if tested for `self.isCommentComposerHidden`.

Reply:
@justinswart question here. do we prefer to isolate the outputs into individual methods or combine them as suggested?

Reply:
Missed this - not sure I understand?

Reply:
I think @mubarak was suggesting to combine the tests for `isCommentComposerHidden` and `configureCommentComposerViewIsBacking` since the name inputs are being called but I separated them to be more explicit. Not sure which is preferable.

Reply:
Oh, yeah I guess that one also isn't named correctly - I meant more in line with how we name the _hidden_ type outputs in the rest of the app. Not a blocker though, we can address later.